pkg/operation/botanist/component/kubeapiserver/kube_apiserver.go

```diff
@@ -285,3 +431,81 @@ func getLabels() map[string]string {
 		v1beta1constants.LabelRole: v1beta1constants.LabelAPIServer,
 	}
 }
+
+// Secrets is collection of secrets for the kube-apiserver.
+type Secrets struct {
+	// BasicAuthentication contains the basic authentication credentials.
+	// Only relevant if BasicAuthenticationEnabled is true.
+	BasicAuthentication *component.Secret
+	// CA is the cluster's certificate authority.
+	CA component.Secret
+	// CAEtcd is the certificate authority for the etcd.
+	CAEtcd component.Secret
+	// CAFrontProxy is the certificate authority for the front-proxy.
+	CAFrontProxy component.Secret
+	// Etcd is the client certificate for the kube-apiserver to talk to etcd.
+	Etcd component.Secret
+	// EtcdEncryptionConfig is the configuration containing information how to encrypt the etcd data.
+	EtcdEncryptionConfig component.Secret
+	// HTTPProxy is the client certificate for the http proxy to talk to the kube-apiserver..
+	// Only relevant if VPNConfig.ReversedVPNEnabled is true.
+	HTTPProxy *component.Secret
+	// KubeAggregator is the client certificate for the kube-aggregator to talk to the kube-apiserver.
+	KubeAggregator component.Secret
+	// KubeAPIServerToKubelet is the client certificate for the kube-apiserver to talk to kubelets.
+	KubeAPIServerToKubelet component.Secret
+	// Server is the server certificate and key for the HTTP server of kube-apiserver.
+	Server component.Secret
+	// ServiceAccountKey is key for service accounts.
+	ServiceAccountKey component.Secret
+	// StaticToken is the static token secret.
+	StaticToken component.Secret
+	// VPNSeed is the client certificate for the vpn-seed to talk to the kube-apiserver.
+	// Only relevant if VPNConfig.ReversedVPNEnabled is false.
+	VPNSeed *component.Secret
+	// VPNSeedTLSAuth is the TLS auth information for the vpn-seed.
+	// Only relevant if VPNConfig.ReversedVPNEnabled is false.
+	VPNSeedTLSAuth *component.Secret
+	// VPNSeedServerTLSAuth is the TLS auth information for the vpn-seed server.
+	// Only relevant if VPNConfig.ReversedVPNEnabled is true.
+	VPNSeedServerTLSAuth *component.Secret
+}
+
+type secret struct {
+	*component.Secret
+	requiredConditionFn func(Values) bool
+}
+
+func (s *Secrets) all() map[string]secret {
+	return map[string]secret{
+		"BasicAuthentication":    {Secret: s.BasicAuthentication, requiredConditionFn: func(v Values) bool { return v.BasicAuthenticationEnabled }},
+		"CA":                     {Secret: &s.CA},
+		"CAEtcd":                 {Secret: &s.CAEtcd},
+		"CAFrontProxy":           {Secret: &s.CAFrontProxy},
+		"Etcd":                   {Secret: &s.Etcd},
+		"EtcdEncryptionConfig":   {Secret: &s.EtcdEncryptionConfig},
+		"HTTPProxy":              {Secret: s.HTTPProxy},
```

Comment:
```suggestion
		"HTTPProxy":              {Secret: s.HTTPProxy, requiredConditionFn: func(v Values) bool { return v.VPN.ReversedVPNEnabled }},
```

Otherwise if ReversedVPN is disabled you get a `missing information for required secret` error.